src/rust_option.rs

```diff
@@ -347,3 +376,152 @@ impl RustOption<*mut core::ffi::c_void> {
         unsafe { &mut *(self as *mut RustOption<*mut core::ffi::c_void> as *mut Option<&'a mut T>) }
     }
 }
+
+#[cfg(feature = "alloc")]
+impl<'a, T> RustOption<&'a RustVec<T>> {
+    pub fn from_option_vec_ref(other: Option<&'a Vec<T>>) -> Self {
+        let _: () = assert_option_safe::<&'a Vec<T>>();
+        match other {
+            None => Self::new(),
+            Some(r) => Self { inner: OptionInner::new(RustVec::from_ref(r))}
+        }
+    }
+
+    pub fn into_option_vec_ref(self) -> Option<&'a Vec<T>> {
+        let _: () = assert_option_safe::<&Vec<T>>();
+        match self.into_value() {
+            None => None,
+            Some(r) => Some(r.as_vec()),
+        }
+    }
+
+    pub fn as_option_vec_ref(&mut self) -> &mut Option<&'a Vec<T>> {
+        let _: () = assert_option_safe::<&Vec<T>>();
+        unsafe { &mut *(self as *mut RustOption<&'a RustVec<T>> as *mut Option<&'a Vec<T>>)}
+    }
+
+    pub fn from_ref_option_vec_ref(v: Option<&Vec<T>>) -> Self {
+        let _: () = assert_option_safe::<&Vec<T>>();
+        Self::from_option_ref(unsafe { core::mem::transmute::<Option<&Vec<T>>, Option<&RustVec<T>>>(v) })
```

Comment:
Changes like these are the scariest to me, but I think they are ok? I was looking at the `RustVec` implementation for guidance: https://github.com/dtolnay/cxx/blob/master/src/rust_vec.rs#L29